### PR TITLE
Add watch to npm scripts

### PR DESCRIPTION
**Pull request by a-tarasyuk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "license": "MIT",
   "scripts": {
     "type-check": "tsc --noEmit",
+    "type-check:watch": "tsc --noEmit --watch",
```
Comment (jednano) [inline package.json]: What about this DRY approach?
```json
"type-check:watch": "npm run type-check -- --watch",
```

**Review by jednano — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -3,10 +3,11 @@
...
   "license": "MIT",
   "scripts": {
-      "type-check": "tsc --noEmit",
-      "build": "npm run build-types && npm run build-js",
-      "build-types": "tsc --emitDeclarationOnly",
-      "build-js": "babel src --out-dir lib --extensions \".ts,.tsx\" --source-maps inline"
+    "type-check": "tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch",
+    "build": "npm run build-types && npm run build-js",
+    "build-types": "tsc --emitDeclarationOnly",
```
Comment (jednano) [inline package.json]: I think this should be `build:types` and `build:js`.

**Review by jednano — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (a-tarasyuk) [reply package.json]: @jedmao https://github.com/Microsoft/TypeScript-Babel-Starter/commit/dcd1285c2c9c6fb6afc5729f183decf8adc81c1a @DanielRosenwasser has changed from ':' to '-'

Comment (jednano) [reply package.json]: Why isn't it `type-check-watch` then? I'm very confused by the inconsistency.

Comment (a-tarasyuk) [reply package.json]: @jedmao btw, you always have a chance to make own PR.

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser why do you prefer to use `-` instead of `:`? Usually, I see that many libraries group related scripts by the prefix `:` - https://github.com/voorhoede/npm-style-guide#group-related-scripts-by-prefix. For instance https://github.com/Microsoft/TypeScript/blob/master/package.json#L100, https://github.com/facebook/create-react-app/blob/next/package.json#L11. Maybe you have a style guide for that, which I can use in order to avoid inconsistency.

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser could you review this PR?

**Comment by DanielRosenwasser** [comment]
Looks good to me - if you want to revert it to colons in another PR, you've convinced me to the point of neutrality on the matter 😄

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser Done. :)

**Pull request merged**